src/payment/strategies/checkoutcom/checkoutcom-google-pay-payment-strategy.ts

```diff
@@ -0,0 +1,109 @@
+import { InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { ContentType } from '../../../common/http-request/';
+import { OrderRequestBody } from '../../../order';
+import Payment from '../../payment';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import { GooglePayPaymentStrategy, PaymentMethodData } from '../googlepay';
+
+import { CheckoutcomGooglePayToken, CheckoutcomToken } from './checkout';
+
+export default class CheckoutcomGooglePayPaymentStrategy extends GooglePayPaymentStrategy {
+    TEST_URL: string = 'https://api.sandbox.checkout.com/tokens';
+    LIVE_URL: string = 'https://api.checkout.com/tokens';
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._googlePayOptions) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.googlepaycheckoutcom" argument is not provided.');
+        }
+
+        const { payment: { methodId = '' } = {} } = payload;
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this._store.getState();
+
+        const testMode: boolean = !!getPaymentMethodOrThrow(methodId).config.testMode;
+        const { checkoutcomkey } = getPaymentMethodOrThrow(methodId).initializationData;
+
+        const {
+            onError = () => {},
+            onPaymentSelect = () => {},
+        } = this._googlePayOptions;
+
+        return Promise.resolve(this._getPayment())
+            .then(payment => {
+                if (!payment.paymentData.nonce || !payment.paymentData.cardInformation) {
+
+                    return this._googlePayPaymentProcessor.displayWallet()
+                        .then(paymentData => this._paymentInstrumentSelected(paymentData))
+                        .then(() => onPaymentSelect())
+                        .then(() => this._getPayment())
+                        .catch(error => {
+                            if (error.statusCode !== 'CANCELED') {
+                                onError(error);
+                            }
+                        });
+                }
+
+            })
+            .then(() => this._convert_token(testMode, checkoutcomkey))
+            .then(payment => {
+                const paymentData: Payment = payment as Payment;
+
+                return this._store.dispatch(this._orderActionCreator.submitOrder({useStoreCredit: payload.useStoreCredit}, options))
+                    .then(() => this._store.dispatch(this._paymentActionCreator.submitPayment(paymentData)));
+            });
+    }
+
+    async requestCheckoutcomTokenize(testMode: boolean | undefined, checkoutcomKey: string, data = {}): Promise<CheckoutcomToken> {
+
+        let url = '';
+        if (testMode) {
+            url = this.TEST_URL;
+        } else {
+            url = this.LIVE_URL;
+        }
+
+        const response = await fetch(url, {
```

Comment:
Can you use `RequestSender`? We don't have polyfills for `fetch` so this won't work in IE11.

Reply:
@davidchin RequestSender forcefully adds and XSRF token to the headers, and causes the request to be rejected by Checkout.com

Reply:
Then I'd say we should modify `RequestSender` to have the option of not passing XSRF header. Perhaps we can do something like this
```ts
requestSender.post(url, data, {
    headers: {
        'Authorization': checkoutcomKey,
        'X-XSRF-TOKEN': null,
    },
});
```

And then in `RequestSender`, we can make it so that we respect the `null` headers that are passed in (https://github.com/bigcommerce/request-sender-js/blob/master/src/request-factory.ts#L33)

```ts
Object.keys(headers)
    .filter(key => headers[key] !== null)
    .forEach(key => {
        xhr.setRequestHeader(key, headers[key]);
    });
```